packages/react-native-editor/src/api-fetch-setup.js

```diff
@@ -1,28 +1,50 @@
 /**
  * WordPress dependencies
  */
-import { fetchRequest } from '@wordpress/react-native-bridge';
+import { fetchRequest, postRequest } from '@wordpress/react-native-bridge';
 import apiFetch from '@wordpress/api-fetch';
 
+const SUPPORTED_METHODS = [ 'GET', 'POST' ];
 // Please add only wp.org API paths here!
-const SUPPORTED_ENDPOINTS = [
-	/wp\/v2\/(media|categories|blocks|themes)\/?\d*?.*/i,
-	/wp\/v2\/search\?.*/i,
-	/oembed\/1\.0\/proxy\?.*/i,
-];
+const SUPPORTED_ENDPOINTS = {
+	GET: [
+		/wp\/v2\/(media|categories|blocks|themes)\/?\d*?.*/i,
+		/wp\/v2\/search\?.*/i,
+		/oembed\/1\.0\/proxy\?.*/i,
+	],
+	POST: [ /wpcom\/v2\/(media)\/.*/i ],
```

Comment:
Based on [this past PR](https://github.com/WordPress/gutenberg/pull/24662), I realized that we should avoid introducing WPCOM endpoints. However, this restriction affects non-core blocks like Jetpack blocks that heavily rely on WPCOM endpoints. Hence, we should refactor this part and probably define/expand the supported endpoints in Gutenberg Mobile.

Reply:
Removal applied in https://github.com/WordPress/gutenberg/pull/49371/commits/b5d54a6315faefdef095c901d8544398abdbbc0b.

Reply:
> I'm thinking to remove this change and work on a separate PR to introduce it via GB-mobile.

@fluiddot, that makes sense to me! I'll go ahead to approve this PR with that understanding 🙇‍♀️